bindinfo/cache.go

```diff
@@ -191,3 +193,84 @@ func (b cache) appendNode(newBindRecord *bindRecord, sparser *parser.Parser) err
 	b[hash] = append(b[hash], newNode)
 	return nil
 }
+
+// AddGlobalBind implements GlobalBindAccessor.AddGlobalBind interface.
+func (h *Handle) AddGlobalBind(originSQL, bindSQL, defaultDB, charset, collation string) error {
+	h.lock.Lock()
+	defer h.lock.Unlock()
+
+	ctx := context.TODO()
+	exec, _ := h.ctx.(sqlexec.SQLExecutor)
+	_, err := exec.Execute(ctx, "BEGIN")
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			_, err = exec.Execute(ctx, "COMMIT")
+		} else {
+			_, rbErr := exec.Execute(ctx, "ROLLBACK")
+			terror.Log(rbErr)
+		}
+	}()
+
+	sql := fmt.Sprintf("SELECT _tidb_rowid,status FROM mysql.bind_info WHERE original_sql='%s' AND default_db='%s'",
+		originSQL, defaultDB)
+	rs, err := exec.Execute(ctx, sql)
+	if err != nil {
+		return err
+	}
+	if len(rs) >= 1 {
+		chkBatch := rs[0].NewRecordBatch()
+		for {
+			err = rs[0].Next(context.TODO(), chkBatch)
+			if err != nil {
+				return err
+			}
+			if chkBatch.NumRows() == 0 {
+				break
+			}
+			it := chunk.NewIterator4Chunk(chkBatch.Chunk)
+			for row := it.Begin(); row != it.End(); row = it.Next() {
+				rowID := row.GetInt64(0)
+				status := row.GetString(1)
+				if status == using {
+					return errors.New("origin sql already has binding sql")
+				}
+				sql = fmt.Sprintf("DELETE FROM mysql.bind_info WHERE _tidb_rowid=%d", rowID)
```

Comment:
I think we can leave this record here.
We can implement a GC goroutine to delete the `deleted records`.

Reply:
This will make bind select gorountine much more complicated. If we need a gc , gc is used to collect the logic delete's bind line.